lib/Common/DataStructures/BaseDictionary.h

```diff
@@ -1066,21 +1064,18 @@ namespace JsUtil
         __ecount(bucketCount) int *AllocateBuckets(DECLSPEC_GUARD_OVERFLOW const uint bucketCount)
         {
             return
-                AllocateArray<AllocatorType, int, false>(
-                    TRACK_ALLOC_INFO(alloc, int, AllocatorType, 0, bucketCount),
-                    TypeAllocatorFunc<AllocatorType, int>::GetAllocFunc(),
+                AllocateArray<TAllocator, int, false>(
+                    TRACK_ALLOC_INFO(alloc, int, TAllocator, 0, bucketCount),
+                    &TAllocator::AllocLeaf,
                     bucketCount);
         }
 
         __ecount(size) EntryType * AllocateEntries(DECLSPEC_GUARD_OVERFLOW int size, const bool zeroAllocate = true)
         {
-            // Note that the choice of leaf/non-leaf node is decided for the EntryType on the basis of TValue. By default, if
-            // TValue is a pointer, a non-leaf allocation is done. This behavior can be overridden by specializing
-            // TypeAllocatorFunc for TValue.
             return
-                AllocateArray<AllocatorType, EntryType, false>(
-                    TRACK_ALLOC_INFO(alloc, EntryType, AllocatorType, 0, size),
-                    zeroAllocate ? EntryAllocatorFuncType::GetAllocZeroFunc() : EntryAllocatorFuncType::GetAllocFunc(),
+                AllocateArray<TAllocator, EntryType, false>(
+                    TRACK_ALLOC_INFO(alloc, EntryType, TAllocator, 0, size),
+                    zeroAllocate ? &TAllocator::AllocZero : &TAllocator::Alloc,
```

Comment:
>zeroAllocate ? &TAllocator::AllocZero : &TAllocator::Alloc, [](start = 20, length = 59)

The removing of `TypeAllocatorFunc` (Allocator.h, Recycler.h) regressed some functionality as shown here. Previously `TypeAllocatorFunc` allows automatic (e.g. int, pointer) or custom (through template specialization) choice of leaf / non-leaf. With the change this functionality is gone. Assume using a BaseDictionary<int, int> on Linux. Since we use RecyclerAllocatorWithBarrier on Linux, then entries will be allocated as non-leaf here. They could be allocated as leaf if TypeAllocatorFunc exists.